### PR TITLE
POW-279

### PR DESCRIPTION
**Pull request by OptimusOpus**
https://dltx.atlassian.net/jira/software/projects/POW/boards/48?selectedIssue=POW-279

image

### CODE REVIEW

File: scripts/deploy/contracts/InvestorVesting.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import { deployContract } from "../utils";
+import { InvestorVesting } from "../../../build/typechain";
+
+export const contractNames = () => ["InvestorVesting"];
+
+export const constructorArguments = () => [
+  "0xd9BAcC5BccAd9A380001d41Cd234b4D5f33ece76",
```
Comment (brettvniekerk) [inline scripts/deploy/contracts/InvestorVesting.ts]: Magic string - what does this do?

**Review by brettvniekerk — APPROVED** [review]
LGTM

File: contracts/InvestorVesting.sol [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+// Client's requirement not yet finalized, still waiting.
+contract InvestorVesting is IInvestor, Ownable {
+    using SafeERC20 for ERC20;
+
+    address public immutable token;
+    uint256 public startDate;
+    uint256 public totalClaimed;
+    uint256 public totalVestingAmount;
+    //Maximum number of vesting details
+    uint256 public maxVestingDetailArray;
```
Comment (xiangzhe0812) [inline contracts/InvestorVesting.sol]: why not just uint256 private constant maxVestingDetailArray = 20

File: contracts/InvestorVesting.sol [hunk 3]
```diff
@@ -0,0 +1,265 @@
...
+
+    address public operator;
```
Comment (xiangzhe0812) [inline contracts/InvestorVesting.sol]: can be private

File: scripts/deploy/contracts/InvestorVesting.ts [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+export const deploy = async (deployer, setAddresses) => {
+  console.log("deploying InvestorVesting");
+  const investorVesting: InvestorVesting = (await deployContract(
+    "InvestorVesting",
+    constructorArguments(),
+    deployer,
+    1
+  )) as InvestorVesting;
+  console.log(`deployed InvestorVesting to address ${investorVesting.address}`);
+  setAddresses({ InvestorVesting: investorVesting.address });
```
Comment (xiangzhe0812) [inline scripts/deploy/contracts/InvestorVesting.ts:22]: can add verify contract command here, like below

import { verifyOnEtherscan } from "../utils";
await verifyOnEtherscan(saleVesting.address, constructorArguments());

**Review by xiangzhe0812 — APPROVED** [review]

**Pull request merged**